src/plugins/WebOfTrust/util/DelayedBackgroundJob.java

```diff
@@ -0,0 +1,26 @@
+package plugins.WebOfTrust.util;
+
+/**
+ * A {@link BackgroundJob} that is executed after triggers have been aggregated for some delay.
+ *
+ * @author bertm
+ */
+public interface DelayedBackgroundJob extends BackgroundJob {
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most some delay
+     * specified by the implementation or as soon as the currently running job finishes, whichever
+     * comes last. Implementations must ensure that this method is safe to invoke from any thread
+     * at any moment.
+     */
+    @Override
+    public void triggerExecution();
```

Comment:
The delay given is in fact treated as a **maximum**, an *upper bound*. The implementation ensures to wait for *at most* the given delay until (trying) executing a job, e.g. it aggregates triggers for *at most* the given delay, in presence of multiple triggers with possibly different delays.
So: the implementation uses the earliest deadline (i.e. the minimum over all deadlines implied by trigger time + delay), so the contract w.r.t. the delays is that they impose a maximum on the earliest deadline (i.e. provide an upper bound).

Consider the following sequence, where each time unit a trigger fires (by default with delay = 10, but once with delay = 5):

```
trigger time   trigger delay   deadline   actual delay
0              10              10         10
1              10              10         9
2              5               7          5
3              10              7          4
4              10              7          3
5              10              7          2
6              10              7          1
--- job execution started/pending at time 7 ---
7              10              17         10
8              10              17         9
9              10              17         8
```

As you can see, the job executes (or is pending to do so) after *at most* the delay given while triggering, so the delay provides a maximum (i.e. upper bound) on the execution delay.